utils/create-node-index-map.ts

```diff
@@ -0,0 +1,11 @@
+import type { SortingNode } from '../typings'
+
+export let createNodeIndexMap = (
+  nodes: SortingNode[],
+): Map<SortingNode, number> => {
+  let nodeIndexMap = new Map<SortingNode, number>()
+  for (let [index, node] of nodes.entries()) {
+    nodeIndexMap.set(node, index)
+  }
+  return nodeIndexMap
+}
```

Comment:
[suggestion] Adding a generic extending `SortingNode`:

```suggestion
export let createNodeIndexMap = <T extends SortingNode>(
  nodes: T[],
): Map<T, number> => {
  let nodeIndexMap = new Map<T, number>()
  for (let [index, node] of nodes.entries()) {
    nodeIndexMap.set(node, index)
  }
  return nodeIndexMap
}
```

Reply:
Added generic.